corehq/apps/users/tasks.py

```diff
@@ -319,5 +326,7 @@ def process_reporting_metadata_staging():
             record.process_record(user)
             record.delete()
 
-    if UserReportingMetadataStaging.objects.exists():
+    duration = datetime.utcnow() - start
+    run_again = run_periodic_task_again(process_reporting_metadata_staging_schedule, start, duration)
+    if run_again and UserReportingMetadataStaging.objects.exists():
```

Comment:
Why not simply keep track of the duration in this function itself and call the update loop till the duration is within specified window? Would avoid the complexity of `run_periodic_task_again`

Reply:
not quite sure what you mean. The intention here is to prevent the task from being recursively called it there is another task that is due to be scheduled via the celery scheduler or if it has passed the window where the task should be run. Only having the duration would work if the task was only triggered once and then left to run recursively until some specified time.

Reply:
Sorry for lack of clarity. I meant instead of recursively calling tasks, would making the record processing loop into a sub function and then keep calling that sub function in a while loop that runs for a certain duration.

Reply:
Ah I see, yes you could do it that way though since the task schedule is configurable you'd still need to interrogate the schedule to figure out what that duration is.

Reply:
> you'd still need to interrogate the schedule to figure out what that duration is.
True, though, the configuration can also be updated to support (start-hour/duration) to simplify this and avoid recursive task explosion. But that sounds a bit out of scope.